onnxruntime/test/shared_lib/test_model_loading.cc

```diff
@@ -4,28 +4,34 @@
 #include "core/session/onnxruntime_cxx_api.h"
 #include "core/platform/env.h"
 #include "onnx_protobuf.h"
-#include <google/protobuf/text_format.h>
 #include "test_fixture.h"
 #include "file_util.h"
+#include <fstream>
+
 namespace onnxruntime {
 namespace test {
 namespace {
 void WriteStringToTempFile(const char* test_data, std::basic_string<ORTCHAR_T>& filename) {
   int fd;
   CreateTestFile(fd, filename);
   onnx::ModelProto mp;
-  if (!google::protobuf::TextFormat::ParseFromString(test_data, &mp)) {
+  if (!mp.ParseFromString(test_data)) {
+    //  if (!google::protobuf::TextFormat::ParseFromString(test_data, &mp)) {
     throw std::runtime_error("protobuf parsing failed");
   }
-  if (!mp.SerializeToFileDescriptor(fd))
+  std::ofstream ofs(filename);
+  std::string out;
+  if (!mp.SerializeToString(&out))
+    //  if (!mp.SerializeToFileDescriptor(fd))
     throw std::runtime_error("write file failed");
+  ofs << out;
   auto st = Env::Default().FileClose(fd);
   if (!st.IsOK())
     throw std::runtime_error("close file failed");
 }
 }  // namespace
 
-TEST_F(CApiTest, model_missing_data) {
+TEST_F(CApiTest, DISABLED_model_missing_data) {  // TODO: protobuf-lite doesn't support parsing protobuf data represented as text
```

Comment:
You don't need to disable this. This program can still link to the full protobuf lib.

Reply:
No, this won't work, since ModelProto is based on proto-lite and hence uses the MessageLite interface.

Reply:
Let this program link to protobuf, not protobuf-lite. This is just a test program.

Reply:
Linking won't solve it because the generated object uses MessageLite as the interface which is a base class of Message.

Reply:
Anyway, who will re-enable this test?

Reply:
Any suggestions on how to fix this test other than creating the text protobuf as binary?

Reply:
If we don't have a solution now, how about this: only disable these tests if protobuf lite is in use.
These tests are important.  I paid a lot of time on them. 

You can exclude this test file in cmake. 
```
if(onnxruntime_USE_FULL_PROTOBUF)
    list(APPEND onnxruntime_shared_lib_test_SRC ${ONNXRUNTIME_SHARED_LIB_TEST_SRC_DIR}/test_tensor_loader.cc)
endif()
```

Reply:
will do.